src/containers/bookmark/Bookmark.js

```diff
@@ -0,0 +1,83 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { Icon } from 'antd';
+import { bindActionCreators } from 'redux';
+import { getBookmarks, addBookmark, removeBookmark } from '../../actions/bookmarks';
+import './bookmark.scss';
+
+export class UnconnectedBookmark extends Component {
+  componentWillMount() {
+    const { getBookmarks } = this.props;
+    getBookmarks();
+  }
+
+  checkBookmark = (bookmarks) => {
+    const { addBookmark, slug, removeBookmark } = this.props;
+    let bookmarkIcon = (
+      <Icon
+        type="book"
+        className="bookmark-container-add-icon"
+        data-test="bookmark-container-add-icon"
+        onClick={() => addBookmark(slug)}
+      />
+    );
+
+    bookmarks.forEach(article => (
+      article.slug === slug
+      && (
+        bookmarkIcon = (
+          <Icon
+            type="book"
+            className="bookmark-container-remove-icon"
+            data-test="bookmark-container-remove"
+            onClick={() => removeBookmark(slug)}
+          />
+        )
+      )
+    ));
+    return bookmarkIcon;
+  }
+
+  displayBookmark = () => {
+    const { bookmarks, slug, addBookmark } = this.props;
+    return Object.getOwnPropertyNames(bookmarks).length !== 0
```

Comment:
What happens when you remove this ` !== 0`

Reply:
When there are no bookmarks, an empty dictionary is returned which will still go through the iteration. the only way to avoid this, I had to check that the dictionary is not empty.